easypy/collections.py

```diff
@@ -847,6 +847,39 @@ def inner(*args, **kwargs):
     return inner
 
 
+def takesome(generator, max=None, min=0):
+    """
+    Take between ``min`` and ``max`` items from the generator.
+    Throws a ValueError if the generator didn't yield ``min`` items.
+
+    >>> list(takesome("abcdef", 3))
+    ['a', 'b', 'c']
+
+    >>> list(takesome("abcdef", 10))
+    ['a', 'b', 'c', 'd', 'e', 'f']
+
+    >>> list(takesome("abcdef", min=8))
+    Traceback (most recent call last):
+        ...
+    ValueError: Not enough items in sequence (wanted 8, got 6)
+    """
+
+    items = [p for p, _ in zip(generator, range(min))]
+
+    if len(items) < min:
+        raise ValueError("Not enough items in sequence (wanted {}, got {})".format(min, len(items)))
+
+    yield from items
+
+    if not max:
```

Comment:
I think `max=0` should yield 0 items, not (potentially) infinite items. This can bite us if we ever do something like:
```python
yield from buffer  # buffer was filled with stuff before
yield from takesome(data_source, chunk_size - buffer)
```
This snippet will work as long as `buffer` does not contain exactly the number of items we want to send in each chunk.

Reply:
what about a couple of uts?

Reply:
The docstrings are doctests...